### PR TITLE
Initial commit for dstack base image

### PR DESCRIPTION
**Pull request by sha-shrestha**

### CODE REVIEW

File: docker/base-image/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  && apt-get install -y libcurl4-openssl-dev \
+  && apt-get install -y libxml2-dev \
+  && pip3 install pyyaml \
+  && pip3 install tqdm \
+  && pip3 install requests \
+  && pip3 install deprecation \
+  && pip3 install -r /requirements.txt \
+  && pip3 install --index-url https://test.pypi.org/simple/ --upgrade --no-cache-dir --extra-index-url=https://pypi.org/simple/ dstack==0.6dev18  \
+  && dstack server update \
+  && nohup bash -c "dstack server start &" \
```
Comment (peterschmidt85) [inline docker/base-image/Dockerfile:28]: I believe dstack server update should be enough. We don’t want to start the server during the build process, do we?

File: docker/base-image/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  && sleep 30 \
+  && ps -ef | grep dstack \
+  && python3 --version \
+  && pip3 --version \
+  && cat /root/.dstack/config.yaml \
+  && python3 /examples/app1.py \
```
Comment (peterschmidt85) [inline docker/base-image/Dockerfile:34]: I suggest that we don’t add examples to the base Docker image. The base one should be empty.

File: docker/base-image/variables.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (peterschmidt85) [inline docker/base-image/variables.sh:1]: I suggest that we set the image name of the base image to dstack

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -54,6 +54,16 @@ jobs:
       - name: Build with Gradle
         run: ./gradlew build
 
+  build-base-image:
```
Comment (peterschmidt85) [inline .github/workflows/build.yml:57]: Should be named publish-docker-snapshot. Should have condition if: github.ref == 'refs/heads/master'

Comment (peterschmidt85) [reply .github/workflows/build.yml:57]: Please also add the instructions how to run docker image to the README.md inside the docker folder.

Comment (peterschmidt85) [reply .github/workflows/build.yml:57]: Also add publish-docker-release that publishes it to the DockerHub. Should have condition that - the tag name (similar to publish-release).

File: docker/base-image/Dockerfile (hunk 1 above)
Comment (sha-shrestha) [reply docker/base-image/Dockerfile:28]: We need to start it if we want to "preload" it with some apps. If no apps are to be preloaded as part of the build process, then I agree there is no need to start it.

File: docker/base-image/Dockerfile (hunk 2 above)
Comment (sha-shrestha) [reply docker/base-image/Dockerfile:34]: Ok

File: docker/base-image/variables.sh (hunk 3 above)
Comment (sha-shrestha) [reply docker/base-image/variables.sh:1]: Ok.

File: .github/workflows/build.yml (hunk 4 above)
Comment (sha-shrestha) [reply .github/workflows/build.yml:57]: Ok.

File: docker/base-image/Dockerfile (hunk 1 above)
Comment (sha-shrestha) [reply docker/base-image/Dockerfile]: Now fixed.

File: docker/base-image/Dockerfile (hunk 2 above)
Comment (sha-shrestha) [reply docker/base-image/Dockerfile]: Fixed.

File: docker/base-image/variables.sh (hunk 3 above)
Comment (sha-shrestha) [reply docker/base-image/variables.sh:1]: Fixed.

**Review by peterschmidt85 — APPROVED** [review]

**Pull request merged**